src/Components/PublicRoute.tsx

```diff
@@ -0,0 +1,34 @@
+import { type FC, type ReactNode, useEffect } from "react";
+import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
+import { Loader } from "@mantine/core";
+
+interface Props {
+  children: ReactNode;
+}
+const PublicRoute: FC<Props> = ({ children }) => {
+  const router = useRouter();
+  const { status } = useSession();
+
+  useEffect(() => {
+    if (status == "authenticated") {
+      void router.push("/notes");
+    }
+  }, [status]);
+  
+  const renderPublicRoute = () => {
+    if (status === "authenticated") {
```

Comment:
Here should probably be status loading, right? :)

Reply:
Yes this works aswell. Gonna change it.